src/browser.qml

```diff
@@ -48,4 +86,45 @@ ApplicationWindow {
         property: "mainWindow"
         value: webView
     }
+
+    Component {
+        id: backgroundComponent
+        Browser.Background {
+            id: bg
+            property Item pageContainer
+            property Page page
+
+            // Page and attached background is destroyed when the page gets destroyed.
+            onPageChanged: {
+                if (!page) {
+                    var index = 0
+                    for (; index < backgrounds.length; ++index) {
+                        if (backgrounds[index].background === bg) {
+                            break
+                        }
+                    }
+
+                    if (index === 0 && backgrounds.length > 0) {
```

Comment:
That would work. Looks that when I wrote this a late night I wasn't thinking too much :)

Index is initialized to 0 but length > 0 guards first block. So, the else block takes care of the rest. Thanks.